booststrap.bat fails in VS console

The `bootstrap.bat` fails to build the tools if preceded with 
```
"C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\vcvarsall.bat" x64
```

The log is:
```
###
### Using 'vc14' toolset.
###

C:\.hunter\_Base\8fe144d\2fea431\3fd9419\Build\Boost\Source\tools\build\src\engine>if exist bootstrap rd /S /Q bootstrap 

C:\.hunter\_Base\8fe144d\2fea431\3fd9419\Build\Boost\Source\tools\build\src\engine>md bootstrap 

C:\.hunter\_Base\8fe144d\2fea431\3fd9419\Build\Boost\Source\tools\build\src\engine>cl /nologo /RTC1 /Zi /MTd /Fobootstrap/ /Fdbootstrap/ -DNT -DYYDEBUG -wd4996 kernel32.lib advapi32.lib user32.lib /Febootstrap\jam0  command.c compile.c constants.c debug.c execcmd.c execnt.c filent.c frames.c function.c glob.c hash.c hdrmacro.c headers.c jam.c jambase.c jamgram.c lists.c make.c make1.c object.c option.c output.c parse.c pathnt.c pathsys.c regexp.c rules.c scan.c search.c subst.c timestamp.c variable.c modules.c strings.c filesys.c builtins.c md5.c class.c cwd.c w32_getreg.c native.c modules/set.c modules/path.c modules/regex.c modules/property-set.c modules/sequence.c modules/order.c 
command.c
compile.c
constants.c
debug.c
execcmd.c
execnt.c
filent.c
filent.c(488): warning C4477: 'sprintf' : format string '%.*s' requires an argument of type 'int', but variadic argument 1 has type '__int64'
filent.c(488): note: this argument is used as a precision
frames.c
function.c
glob.c
hash.c
hdrmacro.c
headers.c
jam.c
jambase.c
jamgram.c
lists.c
make.c
make1.c
object.c
Generating Code...
Compiling...
option.c
output.c
parse.c
pathnt.c
pathsys.c
regexp.c
rules.c
scan.c
search.c
subst.c
timestamp.c
variable.c
modules.c
strings.c
filesys.c
builtins.c
md5.c
class.c
cwd.c
w32_getreg.c
Generating Code...
Compiling...
native.c
set.c
path.c
regex.c
property-set.c
sequence.c
order.c
Generating Code...

C:\.hunter\_Base\8fe144d\2fea431\3fd9419\Build\Boost\Source\tools\build\src\engine>.\bootstrap\jam0 -f build.jam --toolset=vc14 "--toolset-root=C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\..\..\VC\ "  clean 
build.jam:192: in module scope
*** argument error
* rule toolset ( name command .type ? : opt.out + : opt.define * : flags * : linklibs * )
* called with: ( cc C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe : -o  : -D : -s -O :  )
* extra argument (x86)/Microsoft
build.jam:156:see definition of rule 'toolset' being called

C:\.hunter\_Base\8fe144d\2fea431\3fd9419\Build\Boost\Source\tools\build\src\engine>.\bootstrap\jam0 -f build.jam --toolset=vc14 "--toolset-root=C:\Program Files (x86)\Microsoft Visual Studio 14.0\Common7\Tools\..\..\VC\ "  
build.jam:192: in module scope
*** argument error
* rule toolset ( name command .type ? : opt.out + : opt.define * : flags * : linklibs * )
* called with: ( cc C:/Program Files (x86)/Microsoft Visual Studio 14.0/VC/bin/amd64/cl.exe : -o  : -D : -s -O :  )
* extra argument (x86)/Microsoft
build.jam:156:see definition of rule 'toolset' being called

C:\.hunter\_Base\8fe144d\2fea431\3fd9419\Build\Boost\Source\tools\build\src\engine>exit /b 1 
```

--- a/src/engine/build.jam
+++ b/src/engine/build.jam
@@ -184,7 +184,7 @@ toolset borland bcc32 : -e -n : /D
     : $(--python-lib[1]) ;
 ## Generic Unix cc
 if ! $(CC) { CC = cc ; }
-toolset cc $(CC) : "-o " : -D
+toolset cc "$(CC)" : "-o " : -D
     : $(CFLAGS)
     [ opt --release : -s -O ]
     [ opt --debug : -g ]
@@ -343,15 +343,15 @@ toolset vacpp xlc : "-o " : -D
     [ opt --debug : -g -qNOOPTimize -qnoinline -pg ]
     -I$(--python-include) -I$(--extra-include)
     : -L$(--python-lib[1]) -l$(--python-lib[2]) [ if-os AIX : -bmaxdata:0x40000000 ] ;
-    
+
 ## IBM XL C/C++ for Linux (little endian)
 toolset xlcpp xlC :  "-o " : -D
     : -Wno-unused -Wno-format
     [ opt --release : -s  ]
     [ opt --debug : -g -qNOOPTimize -qnoinline -pg ]
     -I$(--python-include) -I$(--extra-include)
-    : -L$(--python-lib[1]) -l$(--python-lib[2]) ;    
-    
+    : -L$(--python-lib[1]) -l$(--python-lib[2]) ;
+
 ## Microsoft Visual C++ .NET 7.x
 toolset vc7 cl : /Fe /Fe /Fd /Fo : -D
     : /nologo

